Transport should be closeable even if not opened

This code should not raise:
```
transport = RequestsTransport()
transport.close()
```

Right now it does:
```python
self = <azure.core.pipeline.transport._requests_basic.RequestsTransport object at 0x0576EEB0>
    def close(self):
        if self._session_owner:
>           self.session.close()
E           AttributeError: 'NoneType' object has no attribute 'close'
```

We didn't see it before, since this bugs happens only if the transport was not used with a context manager, nor saw any requests (both situtation will call "open()", and then there is no bug). 

close() should just do nothing if there is nothing to do.

aiohttp transport should be investigated as well part of this bug

Thanks @rakshith91 for the report ;)

--- a/sdk/core/azure-core/azure/core/pipeline/transport/_requests_basic.py
+++ b/sdk/core/azure-core/azure/core/pipeline/transport/_requests_basic.py
@@ -202,7 +202,7 @@ class RequestsTransport(HttpTransport):
             self._init_session(self.session)
 
     def close(self):
-        if self._session_owner:
+        if self._session_owner and self.session:
             self.session.close()
             self._session_owner = False
             self.session = None

--- a/sdk/core/azure-core/tests/test_basic_transport.py
+++ b/sdk/core/azure-core/tests/test_basic_transport.py
@@ -385,3 +385,7 @@ def test_recursive_multipart_receive():
 
     internal_response0 = internal_response[0]
     assert internal_response0.status_code == 400
+
+def test_close_unopened_transport():
+    transport = RequestsTransport()
+    transport.close()

